packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -56,11 +183,12 @@ export default {
         const restorePath = hasShareJail
           ? buildWebDavSpacesPath(this.space.id, resource.path)
           : buildWebDavFilesPath(this.user.id, resource.path)
+        const overwrite = filesToOverwrite.includes(resource)
 
         restorePromises.push(
           restoreQueue.add(async () => {
             try {
-              await this.$client.fileTrash.restore(path, resource.id, restorePath)
+              await this.$client.fileTrash.restore(path, resource.id, restorePath, overwrite)
```

Comment:
Let's go the extra mile and introduce `restoreDeletedFile` (or similar wording) in `web-client`. We can do that together.

Reply:
done